src/mul/kernel/generic.jl

```diff
@@ -2,6 +2,7 @@
 k: an index from subspace
 indices: array of indices from comspace
 """
+# real U, single real state
```

Comment:
You should restrict `T<:Real` if it can only be applied on real state.
There are similar problems in U and other comments you wrote in this file.

Reply:
`subspace_mul_kernel_generic!` is not limited to real numbers, and it's useful for symbolic computation, among other things. Therefore, my comments were wrong, and I've corrected them.